helix-core/src/main/java/org/apache/helix/controller/stages/TopStateHandoffReportStage.java

```diff
@@ -469,16 +475,17 @@ private void reportTopStateComesBack(ResourceControllerDataProvider cache, Map<S
       }
     }
 
+    long duration = handOffEndTime - handOffStartTime;
+    long helixLatency = duration - fromTopStateUserLatency - toTopStateUserLatency;
+    // It is possible that during controller leader switch, we lost previous master information
+    // and use current time to approximate missing top state start time. If we see the actual
+    // user latency is larger than the duration we estimated, we use user latency to start with
+    duration = Math.max(duration, helixLatency);
+    boolean isGraceful = record.isGracefulHandoff();
+    logMissingTopStateInfo(duration, helixLatency, isGraceful, partition.getPartitionName());
```

Comment:
the if condition below says: ( handOffEndTime - handOffStartTime <= threshold )
- which is not ESPRESSO ask, so why are we moving the message unconditionally?  if it is for improving debuggability, then it will flood as it will log for everything which is missing top state and especially when instance goes down, it will definately cause lot more messages.